File: app/src/main/java/com/georgcantor/wallpaperapp/network/interceptors/ResponseCacheInterceptor.java
package com.georgcantor.wallpaperapp.network.interceptors;

import android.support.annotation.NonNull;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Response;

/**
 * Created by Alex on 20.11.2017.
 */
// Interceptor to cache data and prevent it loading again and again in a minute
public class ResponseCacheInterceptor implements Interceptor {
    @Override
    public Response intercept(@NonNull Chain chain) throws IOException {
        Response originalResponse = chain.proceed(chain.request());

        return originalResponse.newBuilder()
                .removeHeader("Pragma")
                .header("Cache-Control", "public, max-age=" + 60)
                .build();
    }
}

File: app/src/main/java/com/georgcantor/wallpaperapp/network/interceptors/ResponseCacheInterceptor.kt
package com.georgcantor.wallpaperapp.network.interceptors

import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class ResponseCacheInterceptor : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val originalResponse = chain.proceed(chain.request())

        return originalResponse.newBuilder()
                .removeHeader("Pragma")
                .header("Cache-Control", "public, max-age=" + 60)
                .build()
    }
}
